sequencer/src/hotshot_commitment.rs

```diff
@@ -83,10 +83,15 @@ pub async fn run_hotshot_commitment_task(opt: &CommitmentTaskOptions) {
     sequence(max, hotshot, contract).await;
 }
 
-async fn sequence(max_blocks: u64, hotshot: HotShotClient, contract: HotShot<Signer>) {
+async fn sequence(mut max_blocks: u64, hotshot: HotShotClient, contract: HotShot<Signer>) {
     loop {
         if let Err(err) = sync_with_l1(max_blocks, &hotshot, &contract).await {
             tracing::error!("error synchronizing with HotShot contract: {err}");
+            // If we are erroring because of a block gas limit exception, we can try again
+            // with fewer blocks next time
+            if err.to_string().contains("exceeds block gas limit") && max_blocks > 1 {
+                max_blocks /= 2
```

Comment:
This means that we might end up underestimating the maximum number of blocks we can send. I don't think this is really big deal but in the extreme if `max_blocks` is too small we can get in a situation where we can't keep up with HotShot, since HotShot almost always produces multiple blocks per L1 block.

As an optimization we could keep track of like `mut soft_block_limit` and `hard_block_limit`. Whenever we _succeed_ in syncing with L1, increase `soft_block_limit` as long as it doesn't exceed `hard_block_limit`.

Reply:
oh yes, good call